torch/distributed/_tensor/dispatch.py

```diff
@@ -123,45 +116,68 @@ def _operator_dispatch(
     kwargs: Dict[str, object],
     sharding_propagator: ShardingPropagator,
 ) -> Tuple[object, OpSchema, OutputSharding]:
-    # check that we are not getting mixed vanilla and Distributed tensors
-    arg_list, _ = tree_flatten(args)
+    # unwrap the op info from args/kwargs
+    flat_args_list, args_spec = tree_flatten(args)
+    flat_kwargs_list, kwargs_spec = tree_flatten(kwargs)
+    flat_args_schema: List[object] = []
+    flat_local_args: List[object] = []
+    flat_kwargs_schema: List[object] = []
+    flat_local_kwargs: List[object] = []
     mesh = None
-    for arg in arg_list:
-        if isinstance(arg, torch.Tensor) and not isinstance(arg, dtensor.DTensor):
-            raise RuntimeError(
-                f"{op_call}: got mixed torch.Tensor and DTensor, need to convert all"
-                " torch.Tensor to DTensor before calling distributed operators!"
-            )
 
+    for arg in flat_args_list:
```

Comment:
Can we also move this to treemap? Not sure how much perf benefit we can get out of it though.

Reply:
oh got it, good to know!